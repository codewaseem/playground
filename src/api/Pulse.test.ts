jest.mock("axios");

import axios, { AxiosRequestConfig } from "axios";
import PulseApi, { LOGIN_URL, LOGIN_ERROR } from "./Pulse";

const VALID_DATA = {
    userName: "taj@aptask.com",
    password: "Aptask123"
};


(axios as jest.Mocked<any>).mockImplementation(({ url, data, method }: AxiosRequestConfig) => {
    if (url == LOGIN_URL && method == "POST") {
        if (data.userName == VALID_DATA.userName && data.password == VALID_DATA.password) {
            return Promise.resolve({
                data: {
                    data: {
                        email: data.userName
                    }
                }
            });
        } else {
            return Promise.reject();
        }
    }
});


describe("PulseApi", () => {

    let pulseApi = PulseApi.getInstance();


    it("Login: should resolve given correct username and password", async () => {

        let userData = await pulseApi.login(VALID_DATA.userName, VALID_DATA.password);
        expect(axios).toHaveBeenCalledWith({
            url: LOGIN_URL,
            method: 'POST',
            data: {
                userName: VALID_DATA.userName,
                password: VALID_DATA.password
            }
        });
        expect(userData.email).toBe(VALID_DATA.userName);
    });

    it("Login: invalid data should reject", async () => {
        expect(pulseApi.login("badusername", "password")).rejects.toMatchSnapshot()
    });
});