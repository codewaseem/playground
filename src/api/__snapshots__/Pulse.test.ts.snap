// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PulseApi Login: invalid data should reject 1`] = `undefined`;

exports[`PulseApi calling add leave without setting authInfo should reject 1`] = `undefined`;

exports[`PulseApi calling add time without setting authInfo should 1`] = `undefined`;
